tests/test_structures.py

```diff
@@ -17,8 +17,8 @@ def test_structure_manager_widget(structure_data_object):
         editors=[
             awb.BasicStructureEditor(title="Basic Editor"),
         ],
+        input_structure=structure_data_object,
```

Comment:
@yakutovicha the failing test indicates (thanks to pytest warning turned into error) that this change is not correct, probably `input_structure` cannot be passed in the constructor?

I am planning to go through the code today (visual testing has been done by others already)

Reply:
Thanks, @danielhollas, I will have a closer look!

Reply:
this has been resolved in 374128a517af85c52e310807f8a69710cf5e4d67

Reply:
@yakutovicha I don't think your solution in that commit is correct (it looks like a breaking change possibly). `input_structure` cannot be passed via the constructor. I think you just need to revert the change in this test.
Unless I am missing something.

Reply:
Let me explain why I did it this way.

1. The `structure` trait is the one that should be the only trait that is linked between the manager and the viewer. It represents the current molecule (possibly with modifications) that user is supposed to see.
2. The `input_structure` trait is the one receiving the structure from importers, it doesn't make sense to pass it to the viewer.
3. Usually, I pass `**kwargs` to the parent class, because they might contain some styling properties. In that case, styling of the manager has very little to do with the styling of the viewer - they are just different.

Based on this I decided not to pass `**kwargs` at all. If one wants to update styling, one can directly provide this info here:

```python
self.viewer = StructureDataViewer(style=..., layout=...)
```

Reply:
Right, that all makes sense, thanks! However, it still seems to me that this change in the test needs to be reverted, no? Looking at the StructureManagerWidget code, passing `input_structure` via constructur will have no effect, since it would be an keyword argument. Am I still missing something?

Reply:
Oh, I think I am getting now what is the problem.

When dealing with traits it is ok to pass a trait through the constructor. They get initialised here:
https://github.com/aiidalab/aiidalab-widgets-base/blob/5abbf5d601879bd50f033f8900caf7d221d9a080/aiidalab_widgets_base/structures.py#L134

Here is the smallest example you can play with:
```python
from traitlets import HasTraits, Int

class Person(HasTraits):
    age = Int(allow_none=True)

# Creating an instance of the Person class
person = Person(age=30)

# Accessing and modifying attributes
print(person.age)  # Output: 30

person.age = 35

print(person.age)  # Output: 35

```

Reply:
CC @edan-bainglass

Reply:
> since sometimes the traitlets are in fact explicitly defined in the constructor.

This also I noticed and I think we should avoid doing it.